### PR TITLE
Use payment gateway suggestion title instead of local plugin names

### LINKED ISSUE #7188: Payment gateway setup install step missing plugin names
**Describe the bug**
The install step used to show the plugin names.  Only those store in the WooCommerce Admin client will show since we don't have access to the plugin name prior to install.

We should either a) add missing strings to [the plugin names list](https://github.com/woocommerce/woocommerce-admin/blob/25a8237cb3bd6931d324ae0f4848335e8fe9d78a/packages/data/src/plugins/constants.ts#L12) or b) use a more generic message like "Install plugins."

https://github.com/woocommerce/woocommerce-admin/blob/25a8237cb3bd6931d324ae0f4848335e8fe9d78a/client/task-list/tasks/payments/PaymentGatewaySuggestions/components/Setup/Setup.js#L90-L92

**To Reproduce**

1. Enable the new payment gateway suggestions feature.
2. Deactivate Paystack if installed.
2. Visit the payments task.
3. Visit the paystack gateway task.
4. Note the "Install" content step.

**Expected behavior**
Expected to see the plugin name(s) or maybe update this string to "Install plugins."

**Screenshots**

Screen Shot 2021-06-15 at 8 39 20 AM
Screen Shot 2021-06-15 at 8 36 36 AM

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #7188 

Uses the gateway suggestion title for installation text

The previous method was to store all plugin names in the data store, but if one was missing, the text would ready "Install " instead of "Install {pluginName}."

Note that the current method does not actually show the plugin name (or multiple plugin names) but instead shows the suggestion title text from the REST API.  I think this is a better solution as it's less prone to becoming out of date.

### Screenshots

Screen Shot 2021-06-29 at 1 35 16 PM

### Detailed test instructions:

1. Navigate to the payment gateway task.
2. Note that the install step contains the payment gateway task title.

### CODE REVIEW

**Review by louwie17 — APPROVED** [review]
Approving this as it tested well and we already use title for the configure step.

I did notice some of our other payment provider titles are longer, like Stripe - `Credit cards - powered by Stripe`, I wonder if we should add an alternate option for a title in those cases? Have a plugin `name` param maybe (given title is used on the payments screen)?

I think only Stripe and the Mercado plugin titles are long.

Screen Shot 2021-07-05 at 2 48 38 PM

cc: @elizaan36

**Comment by mattsherman** [comment]
> I did notice some of our other payment provider titles are longer, like Stripe - `Credit cards - powered by Stripe`, I wonder if we should add an alternate option for a title in those cases? Have a plugin `name` param maybe (given title is used on the payments screen)?

That would be ideal. Perhaps in addition we could show the title italicized?

These refinements, if @elizaan36 agrees, could be done in a follow-up PR, so that we can wrap this one up.

**Comment by joshuatf** [comment]
Great points @louwie17 and @mattsherman!  Will wait for @elizaan36 to weigh on here before merging this one in.

**Comment by elizaan36** [comment]
It would be best to have just "[gateway name] account" rather than the long title. So ideally it would be "Configure your Stripe account" or if that's a problem "Configure your _Credit cards powered by Stripe_ account". The dash is really confusing in the title so even just removing that would be an improvement.

**Comment by joshuatf** [comment]
Those names are all taken from the existing list payment gateway suggestions list, so I'm not sure if there are any contractual obligations.  cc @mattsherman @pmcpinto if you have any clues here.

Stripe is the only oddball in this list that stands out.  I think it would be great to update the gateway title to `Stripe` instead of `Credit cards - powered by Stripe` if there aren't any issues with that.

We could introduce a new prop here purely for the plugin name, but I think that complicates the component and REST APIs needlessly.

**Comment by mattsherman** [comment]
> Those names are all taken from the existing list payment gateway suggestions list, so I'm not sure if there are any contractual obligations. cc @mattsherman @pmcpinto if you have any clues here.

I am not familiar with the contractual obligations, so I'll definitely defer to @pmcpinto here for guidance.

> Stripe is the only oddball in this list that stands out. I think it would be great to update the gateway title to `Stripe` instead of `Credit cards - powered by Stripe` if there aren't any issues with that.

That would be so much more readable. 

> We could introduce a new prop here purely for the plugin name, but I think that complicates the component and REST APIs needlessly.

Agreed.

**Comment by pmcpinto** [comment]
> Those names are all taken from the existing list payment gateway suggestions list, so I'm not sure if there are any contractual obligations. cc @mattsherman @pmcpinto if you have any clues here.

@joshuatf AFAIK we don't have any obligation in regards to the copy that we should use in this part of the flow. But I'm pretty confident that all partners prefer to just have their gateway name: Install Stripe / Configure your Stripe account.

**Comment by joshuatf** [comment]
Thanks, @pmcpinto!  Updated to `Stripe`.

**Pull request merged**
